### PR TITLE
Module organisation

### LINKED ISSUE #395: Module organisation
I want to look into our API documentation, but we first have to decide if we want to change the module organisation. 

We've added quite a few modules / changed code and I think we should re-think how we organize them (minor modification only).

These are my suggestions for changes:

- new `configuration` directory including configuration.py, commandline_parser.py
- new 'data_model' directory including data_model.py, model_data_writer.py, validate_data.py, validate_schema.py, workflow_description.py
- `job_submission` - suggest to rename it to `job_execution` or `execution`. There is only one module in this directory, but I would like to add one more (which separates calling any external software from the rest of the code and allows for easier integration of containers)
- `util/model.py` is a bit confusing, as there is also a model directory. I suggest to rename it and move it to model/model_utils.py

### PR DESCRIPTION
**Pull request by GernotMaier**
This PR implements the module organisation as discuss in issue #395.

We will have to deal with the documentation in a separate PR (most of it should be done automatically)

I repeat here the re-organization outlined in the issue:

- new `configuration` directory including configuration.py, commandline_parser.py
- new 'data_model' directory including data_model.py, model_data_writer.py, validate_data.py, validate_schema.py, workflow_description.py
- `job_submission` - rename it to `job_execution`
- `util/model.py`  rename it and move it to model/model_utils.py

Closes #395

### CODE REVIEW

File: applications/add_file_to_db.py [hunk 1]
```diff
@@ -37,7 +37,7 @@
 import logging
 from pathlib import Path
 
-import simtools.configuration as configurator
+import simtools.configuration.configurator as configurator
```
Comment (orelgueta) [inline applications/add_file_to_db.py:40]: This should be instead 

`from simtools.configuration import configurator`

File: simtools/data_model/validate_schema.py [hunk 2]
```diff
@@ -2,7 +2,7 @@
 import logging
 import re
 
-import simtools.util.data_model as data_model
+import simtools.data_model.data_model as data_model
```
Comment (orelgueta) [inline simtools/data_model/validate_schema.py:5]: same issue as `configurator`

**Review by orelgueta — COMMENTED** [review]
A few minor issues

File: applications/submit_data_from_external.py [hunk 3]
```diff
@@ -35,11 +35,11 @@
 import logging
 from pathlib import Path
 
+import simtools.data_model.model_data_writer as writer
+import simtools.data_model.validate_data as ds
```
[the remark this answers is not in the document]
Comment (GernotMaier) [reply applications/submit_data_from_external.py:39]: Because it is shorter! But I changed it.

**Review by orelgueta — APPROVED** [review]
All good now!

**Pull request merged**